### ISSUE TITLE
Application crashes when attempting to save diagram

### ISSUE THREAD

**Issue by aunsbjerg** [gaphor/gaphor#2015]
### Describe the bug
Application crashes when trying to save using `ctrl+s`.

### To Reproduce
1. open program and open a diagram
2. press ctrl+s

This happens regardless of whether using `GDK_BACKEND=x11` or not.

### Expected behavior
I would expect a save dialog

### Screenshots

Log from running the application:

```bash
╰─ GDK_BACKEND=x11 ./Gaphor-2.15.0-x86_64.AppImage                           ─╯
gaphor.storage.storage INFO Read 58 elements from file

(gaphor-exe:20707): GtkSourceView-WARNING **: 10:47:41.343: Failed to load '/tmp/.mount_GaphorrmXmlY/gaphor/ui/language-specs/gaphorcss.lang': could not find the RelaxNG schema file

(gaphor-exe:20707): GLib-GIO-ERROR **: 10:48:34.861: Settings schema 'org.gtk.gtk4.Settings.FileChooser' is not installed
[1]    20707 trace trap (core dumped)  GDK_BACKEND=x11 ./Gaphor-2.15.0-x86_64.AppImage
```

And error report from the tools menu:

```
Gaphor version:    2.15.0
Operating System:  Linux (5.14.0-1055-oem)
Python version:    3.11.1
GTK version:       4.8.4
Adwaita version:   1.2.1
PyGObject version: 3.43.0
Pycairo version:   1.23.1
Cairo version:     1.17.7
Pango version:     1.50.13
Display:           GdkX11Display

No errors have been intercepted. We’re good.
```

### OS
- [x] Linux (Please put in notes the specific distro)
- [ ] macOS
- [ ] Windows

### Version
Version of Gaphor: Gaphor-2.15.0-x86_64.AppImage

### Additional information
Seems to be identical to #1355 which was apparently resolved in a later version.

**Comment by amolenaar** [comment]
Hi @aunsbjerg,

Thank you for reporting this issue. What linux distribution are you running? Do you have GTK4 installed on your host machine?

**Comment by aunsbjerg** [comment]
> Thank you for reporting this issue. What linux distribution are you running? 

I am using Ubuntu 20.04.5 LTS

> Do you have GTK4 installed on your host machine?

Not to my knowledge. Is GTK4 a pre-requisite for gaphor?

**Comment by amolenaar** [comment]
It should not have to be. I've created a PR. I think I nailed the issue, since I could make it fail in CI with a similar error.

If you can try the AppImage produced by one of it's builds, that's much appreciated.
https://github.com/gaphor/gaphor/actions/runs/3950433536

**Comment by aunsbjerg** [comment]
> It should not have to be. I've created a PR. I think I nailed the issue, since I could make it fail in CI with a similar error.
> 
> If you can try the AppImage produced by one of it's builds, that's much appreciated. https://github.com/gaphor/gaphor/actions/runs/3950433536

Couldn't get your link to work, but I'm guessing you are referring to [this](https://github.com/gaphor/gaphor/suites/10444183404/artifacts/516851811) artifact? I tested with that and that works as expected, a save dialog pops up on `ctrl+s` and I can save the diagram. Thanks for looking into this!

**Comment by danyeaw** [comment]
@all-contributors please add @aunsbjerg for bug

**Comment by danyeaw** [comment]
Thanks @aunsbjerg!

Closed by #2016

**Comment by amolenaar** [comment]
From what I read from the (Glib) code, setting XDG_DATA_DIRS is the right thing to do.

The XDG_DATA_DIRS environment variable is properly set in the [glib runtime hook of PyInstaller](https://github.com/pyinstaller/pyinstaller/blob/develop/PyInstaller/hooks/rthooks/pyi_rth_glib.py#L21).
So my patch https://github.com/gaphor/gaphor/pull/2016 is not (completely) correct. As far as I know, setting os.environ also propagates to the C layer.
Maybe the resources are not packaged because we build stuff with JHBuild?
Edit: the schema's are part of the AppImage

@danyeaw Any ideas?

**Comment by amolenaar** [comment]
https://github.com/pyinstaller/pyinstaller/issues/7394

**Comment by amolenaar** [comment]
Fixed by #2020. Could be improved by a patch on PyInstaller.
